Merge from upstream v13

Alchez opened the pull request:
**Changelogs:**

- https://github.com/frappe/frappe/releases/tag/v13.26.0
- https://github.com/frappe/frappe/releases/tag/v13.26.1
- https://github.com/frappe/frappe/releases/tag/v13.27.0
- https://github.com/frappe/frappe/releases/tag/v13.28.0
- https://github.com/frappe/frappe/releases/tag/v13.29.0
- https://github.com/frappe/frappe/releases/tag/v13.29.1
- https://github.com/frappe/frappe/releases/tag/v13.29.2
- https://github.com/frappe/frappe/releases/tag/v13.30.0

phot0n committed:
feat(minor): use specific email signature via from field

* fix: increase debounce timeout so that events dont clash

* fix: signature getting appended multiple time upon dialog open/close

(cherry picked from commit b9af0ac37e47cc54bc77aa97794927ba2bb51aeb)

# Conflicts:
#	frappe/public/js/frappe/ui/field_group.js

gavindsouza committed:
fix(Kanban Board): Clear user settings & dpctype cache on_change

Deletion of kanban boards may lead to 404s on clicking on the Kanban
option in the views. This is due to the cache not being synced with the
DB changes. Changed the hook to on_change since it gets triggered even
in db_set.

(cherry picked from commit fd0564953b0d5dd32d563617b1894e48d0aa8c55)

gavindsouza committed:
fix: Don't update saved filters "silently"

Blame on the filters would not be maintained since update_modified was
unset. Also, this API allowed those without access to Kanban Boards to
possibly pass "bad" filters and cause system to fail (possible DOS).

Using client's set_value does a better job at tackling this.

(cherry picked from commit 540b34347c688b32ecc5538296f4c6a95dbab1ae)

# Conflicts:
#	frappe/desk/doctype/kanban_board/kanban_board.py

gavindsouza committed:
fix: Check if kanban exists before switching route

(cherry picked from commit 87c6cbeee2845e0b73b323afcf53c3d5143996e2)

gavindsouza committed:
refactor: new Kanban dialog

* Hide 'Save' button if no select fields exist
* Show 'Customize Form' as primary button when no selects exist

(cherry picked from commit 13922966996d1013a3dcbdf937009800ef21adf5)

shariquerik committed:
fix: added add signature button if multiple sender emails available

(cherry picked from commit 0a1595e94e5bd1af7c44bcf4fbd45f3d14438100)

shariquerik committed:
chore: better variable name

(cherry picked from commit 7679d5e82cc5e596ba78075649f211f7da81090f)

mergify[bot] committed:
feat: add "After Insert" doctype event to server script (backport #16543) (#16556)

Co-authored-by: kamaljohnson <kamaljohnson12345@gmail.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

nabinhait committed:
fix: Show assignments based on allocated todo only (#16535)

mergify[bot] committed:
fix: new route syntax for Logs in System Console (backport #16554) (#16557)

Co-authored-by: Faris Ansari <netchamp.faris@gmail.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

mergify[bot] committed:
fix: stabilize leaflet map (backport #16513) (#16560)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

shariquerik committed:
style: Added gap between from and to field

(cherry picked from commit 1e63475b2ae6bfced2a473f7554f8963b2b43681)

shariquerik committed:
fix: resolved conflicts

shariquerik committed:
fix: reverting arrow function to normal function

mergify[bot] committed:
Merge pull request #16569 from frappe/mergify/bp/version-13-hotfix/pr-16507

feat(minor): use specific email signature via from field (backport #16507)

mergify[bot] committed:
fix: update french translation (backport #16550) (#16571)

Co-authored-by: HENRY Florian <florian.henry@open-concept.pro>

surajshetty3416 committed:
ci: Add isort & black to pre-commit config

frappe-bot committed:
style: Format codebase & sort imports

surajshetty3416 committed:
chore: ignore bulk formatting commit from git-blame

hans-permana committed:
fix: use user as the filter value instead of modified (#16586)

(cherry picked from commit 08b87f517bc1e9828a2b32b7c14216944a2c397b)

gavindsouza committed:
fix(desk): Use Int control for Long Int (#16572)

Type `Long Int` doesn't have a JS control, we can use the Int's instead. Prior to this, forms of DocTypes with Long Int fields would just not render :trollface:

<img width="1440" alt="Screenshot 2022-04-11 at 3 36 40 PM" src="https://user-images.githubusercontent.com/36654812/162716347-c924775c-50ee-4e8c-8627-21c178f4e221.png">

(cherry picked from commit 18108eef65389faa90d824e243419937d4e3e7ad)

# Conflicts:
#	frappe/public/js/frappe/form/controls/int.js

gavindsouza committed:
Merge pull request #16589 from frappe/mergify/bp/version-13-hotfix/pr-16572

fix(desk): Use Int control for Long Int (backport #16572)

gavindsouza committed:
Merge pull request #16588 from frappe/mergify/bp/version-13-hotfix/pr-16586

fix: use the right filter value when flushing route history (backport #16586)

surajshetty3416 committed:
Merge branch 'version-13-hotfix' of https://github.com/frappe/frappe into format-python

mergify[bot] committed:
feat: Ignore permlevel for specific fields (backport #16590) (#16592)

Co-authored-by: Nabin Hait <nabinhait@gmail.com>

deepeshgarg007 committed:
Merge pull request #16594 from frappe/version-13-hotfix

chore: Pre release for v13.26.0

surajshetty3416 committed:
Merge pull request #16583 from surajshetty3416/format-python

deepeshgarg007 committed:
chore: change log and version bump

deepeshgarg007 committed:
chore: Add change log

deepeshgarg007 committed:
Merge pull request #16598 from frappe/version-13-pre-release

chore: Release for v13.26.0

deepeshgarg007 committed:
Merge pull request #16597 from deepeshgarg007/v13_26_0

chore: change log and version bump

mergify[bot] committed:
fix: Email Dialog 'To' field is not full width (backport #16603) (#16606)

Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>

gavindsouza committed:
fix: Don't port breaking changes from develop

gavindsouza committed:
fix!: Allow child table naming

(cherry picked from commit 03d542edce8dd87f956c26d583180e49126ba683)

gavindsouza committed:
Merge pull request #16609 from frappe/mergify/bp/version-13-hotfix/pr-16608

fix!: Allow child table naming (autoname) (backport #16608)

surajshetty3416 committed:
Merge pull request #16607 from frappe/mergify/bp/version-13-hotfix/pr-16509

ankush committed:
chore: check in missing __init__.py (#16617)

FHenry committed:
fix: update translation (#16623)

(cherry picked from commit 0ae7248847399e5ef3499afea943398d1c715e38)

mergify[bot] committed:
Merge pull request #16627 from frappe/mergify/bp/version-13-hotfix/pr-16623

fix: update translation (backport #16623)

nikkothari22 committed:
fix: Fixed spelling of "Submitting" in Bulk Upload Modal (#16638)

* Fixed spelling of "Submitting"

* Updated translation files

(cherry picked from commit 746e017aeff05d26a6cc87ed31aa460cd2accc47)

# Conflicts:
#	frappe/desk/doctype/bulk_update/bulk_update.py

ankush committed:
fix: conflicts

bhavesh95863 committed:
fix: color hex code not work as class name

surajshetty3416 committed:
refactor: Move checkbox style to a separate file to re-use it in print

(cherry picked from commit 155dbb5652c0b558a25d7b1843777130ccb57d50)

# Conflicts:
#	frappe/public/scss/print.scss

surajshetty3416 committed:
fix: Resolve conflicts

surajshetty3416 committed:
Merge pull request #16646 from bhavesh95863/patch-6

surajshetty3416 committed:
Merge pull request #16639 from frappe/mergify/bp/version-13-hotfix/pr-16638

fix: Fixed spelling of "Submitting" in Bulk Upload Modal (backport #16638)

mergify[bot] committed:
fix: Update default currency & date format for Malta. (backport #16630) (#16651)

gavindsouza committed:
fix: Set 'Attached to *' fields properly

Check if DocType is a child table before using parent and parenttype
attributes. This caused problems when a dead "parent" column existed on
the database and was mapped to the Document object.

(cherry picked from commit 27d72ce017e2ab99b09d78c144e468543c06986a)

gavindsouza committed:
chore: Simplify logic in extract_images_from_html

(cherry picked from commit 6770e4d89d5ffa7def8b4060c6725179bc7adfe2)

mergify[bot] committed:
fix: extend autoname validation to child items (backport #16624) (#16625)

* fix: extend autoname validation to child items

(cherry picked from commit b409a7eeba7f1ad249ec1697ec4fae70def44152)

* test: naming on child tables

(cherry picked from commit a615c60be44efe5d49fbdd4df37d87366b14ab09)

* test: accept doctype field values in `new_doctype`

Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
Merge pull request #16652 from frappe/mergify/bp/version-13-hotfix/pr-16648

mergify[bot] committed:
chore: Add semantic releases (backport #16647) (#16658)

* chore: Add semantic releases

(cherry picked from commit bafb1faa4891c9be76c3d12aa4f13cbe48ab4b88)

* chore: block major releases

(cherry picked from commit 6c7d867dbd4e8dc83f46a958fccc166e5850c681)

* chore: do not publish any assets

(cherry picked from commit 17aeec04c85168e84d94dc8205134a959f74cedd)

* chore: use ubuntu latest

(cherry picked from commit bf21d2fe2a3dee9ec68556a05ff795c2ea5c1779)

Co-authored-by: Deepesh Garg <deepeshgarg6@gmail.com>

mergify[bot] committed:
fix: Set dependant property of grid fields while rendering (backport #16548) (#16656)

mergify[bot] committed:
fix: kanban board without title field (#16659) (#16660)

(cherry picked from commit 7b19c91c373c3820f7a3c029b1df25ca42b0a913)

Co-authored-by: Ankush Menat <ankush@frappe.io>

gavindsouza committed:
Merge pull request #16657 from frappe/mergify/bp/version-13-hotfix/pr-16654

fix: Link extracted images from documents correctly [v13] (backport #16654)

deepeshgarg007 committed:
Merge branch 'version-13-pre-release' into version-13-hotfix

deepeshgarg007 committed:
Merge pull request #16667 from frappe/version-13-hotfix

chore: Pre-relese for v13.27.0

deepeshgarg007 committed:
Merge branch 'version-13' into version-13-pre-release

deepeshgarg007 committed:
Merge pull request #16668 from frappe/version-13-pre-release

chore: release for v13.27.0

semantic-release-bot committed:
chore(release): Bumped to Version 13.26.1

## [13.26.1](https://github.com/frappe/frappe/compare/v13.26.0...v13.26.1) (2022-04-19)

### Bug Fixes

* Check if kanban exists before switching route ([b1f9443](https://github.com/frappe/frappe/commit/b1f9443c386913b0ae13353c769d001d51734cbd))
* color hex code not work as class name ([4ff474d](https://github.com/frappe/frappe/commit/4ff474d0883b356bc498cb62e7e4ff591e387429))
* conflicts ([97fff7e](https://github.com/frappe/frappe/commit/97fff7e19031db4e95aca657bfe42bbe515883bd))
* Don't port breaking changes from develop ([4afbfa5](https://github.com/frappe/frappe/commit/4afbfa553086bc544ec3d1110521e8c65e215a3d))
* Don't update saved filters "silently" ([b143155](https://github.com/frappe/frappe/commit/b143155843fefa94347b906d258b048794512e42))
* Email Dialog 'To' field is not full width (backport [#16603](https://github.com/frappe/frappe/issues/16603)) ([#16606](https://github.com/frappe/frappe/issues/16606)) ([905fd99](https://github.com/frappe/frappe/commit/905fd99b5fff8ca851c957be532e98b7f7be35c8))
* extend autoname validation to child items (backport [#16624](https://github.com/frappe/frappe/issues/16624)) ([#16625](https://github.com/frappe/frappe/issues/16625)) ([649e3ce](https://github.com/frappe/frappe/commit/649e3ce541d3843cae1d99585fdc7dd90895e4a7))
* Fixed spelling of "Submitting" in Bulk Upload Modal ([#16638](https://github.com/frappe/frappe/issues/16638)) ([8077132](https://github.com/frappe/frappe/commit/80771329f26bfc61f4d7cb6545579b1d15d81cca))
* kanban board without title field ([#16659](https://github.com/frappe/frappe/issues/16659)) ([#16660](https://github.com/frappe/frappe/issues/16660)) ([3dce038](https://github.com/frappe/frappe/commit/3dce038cdec1f6fdd4d34470fca4b88e48ada805))
* **Kanban Board:** Clear user settings & dpctype cache on_change ([6b2aa98](https://github.com/frappe/frappe/commit/6b2aa987c9a12d76dd64dbdb146a4de7cb3782e5))
* Resolve conflicts ([21c9f75](https://github.com/frappe/frappe/commit/21c9f75320f68d99ae48f04a2ca12ae2bd49f984))
* Set 'Attached to *' fields properly ([774e882](https://github.com/frappe/frappe/commit/774e8827a439194dae75e9d11b0dca0eb460cdbd))
* Set dependant property of grid fields while rendering (backport [#16548](https://github.com/frappe/frappe/issues/16548)) ([#16656](https://github.com/frappe/frappe/issues/16656)) ([3ef42e6](https://github.com/frappe/frappe/commit/3ef42e605766bbdc0a9949ae04038932ce1f1c5b))
* Update default currency & date format for Malta. (backport [#16630](https://github.com/frappe/frappe/issues/16630)) ([#16651](https://github.com/frappe/frappe/issues/16651)) ([0e75c17](https://github.com/frappe/frappe/commit/0e75c177d2d3220025a1bf2561af22179829256a))
* update translation ([#16623](https://github.com/frappe/frappe/issues/16623)) ([e1f5e16](https://github.com/frappe/frappe/commit/e1f5e16178e6eb52ee1b072ac652edffd81890f2))

mergify[bot] committed:
chore: Update creds to allow updates on protected branch (#16672) (#16673)

(cherry picked from commit b59619f764a09110e8a8a07dff5d4bc6352a4746)

Co-authored-by: Deepesh Garg <42651287+deepeshgarg007@users.noreply.github.com>

gavindsouza committed:
fix: Handle unable to read commit errors

(cherry picked from commit aef0c0f66e4b1997ef946998e6822de7a8ee58cf)

gavindsouza committed:
Merge pull request #16677 from frappe/mergify/bp/version-13-hotfix/pr-16675

fix(version): Handle unable to read commit errors in version command (backport #16675)

mergify[bot] committed:
feat: add/remove fields from kanban board (backport #16257) (#16671)

* feat: add/remove fields from kanban board (#16257)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>
Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>
Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>

mergify[bot] committed:
feat(minor): add ignore_xss_filter to customize form field (backport #16612) (#16674)

mergify[bot] committed:
test: Typing on focused input issue (backport #16669) (#16670)

mergify[bot] committed:
fix: Check mandatory_depends_on also while checking mandatory on save (#16515) (#16686)

surajshetty3416 committed:
fix: Do not change direction of text if already in RTL mode

(cherry picked from commit 56bb11dc17ed16de577263908e51ea6d076ea44f)

surajshetty3416 committed:
Merge pull request #16694 from frappe/mergify/bp/version-13-hotfix/pr-16693

fix: Do not change direction of text if already in RTL mode (backport #16693)

mergify[bot] committed:
fix: handle no data in set_open_count (backport #16582) (#16696)

mergify[bot] committed:
fix(UI): Duration Filter overlapping issue  (backport #16599) (#16703)

Co-authored-by: KrutikaBhatt <65107474+KrutikaBhatt@users.noreply.github.com>
Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>

ankush committed:
feat: configurable auto email reports limit (#16684)

- The previous limit was 3 per user which is way too less, no known reason to
restrict this other than hogging of system with too many reports. Bumped
default limit to 20.
- site config is not easily discoverable or editable, added config in
  system settings.
- Moved auto email report background job form daily queue to `daily_long` queue.

closes https://github.com/frappe/frappe/issues/16681

<img width="1049" alt="Screenshot 2022-04-20 at 12 33 06 PM" src="https://user-images.githubusercontent.com/9079960/164170117-5612d9df-da91-441b-a4f6-acd89d30336e.png">

`no-docs` (error message is sufficient to explain to user what to do without referring docs)

ref:

ISS-21-22-10245
ISS-21-22-07742
ISS-20-21-10850
ISS-20-21-10112

and many more times. This is such a stupid validation 🤦

(cherry picked from commit a245cb51a24f044ce9e332c7193c16abfa21d325)

ankush committed:
Merge pull request #16714 from frappe/mergify/bp/version-13-hotfix/pr-16684

feat: configurable auto email reports limit (backport #16684)

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

ghost commented:
### Features

* add "After Insert" doctype event to server script (backport #16543) (#16556) ([0685dc0](https://github.com/ParsimonyGit/frappe/commit/0685dc0709771c02abb30ebc749c88398d4c32be))
*  **minor:** use specific email signature via from field ([a175f95](https://github.com/ParsimonyGit/frappe/commit/a175f95b24ac2e93e23b473ca83b80e9b9680164))
* Ignore permlevel for specific fields (backport #16590) (#16592) ([c43084b](https://github.com/ParsimonyGit/frappe/commit/c43084b106faf1bf92f465a669e2abab1d3bd7d3))


### Bug Fixes

* Show assignments based on allocated todo only (#16535) ([1e6532c](https://github.com/ParsimonyGit/frappe/commit/1e6532c841fcb4e605c919287178f019ad06d0ee))
* new route syntax for Logs in System Console (backport #16554) (#16557) ([1cd5148](https://github.com/ParsimonyGit/frappe/commit/1cd5148af29fc6867ca6bc9aa2965a7619bf85d6))
* stabilize leaflet map (backport #16513) (#16560) ([5191368](https://github.com/ParsimonyGit/frappe/commit/51913688658be64b78e44a715c279c325066fa3b))
* added add signature button if multiple sender emails available ([4ef589f](https://github.com/ParsimonyGit/frappe/commit/4ef589f928506a8214896f75b09d7a02f6b84208))
* resolved conflicts ([f9061c4](https://github.com/ParsimonyGit/frappe/commit/f9061c40d9b1a091ed63b856fbeec504273ad03a))
* reverting arrow function to normal function ([624e158](https://github.com/ParsimonyGit/frappe/commit/624e15869fdf5714a087f4843481c5985124c491))
* update french translation (backport #16550) (#16571) ([2703569](https://github.com/ParsimonyGit/frappe/commit/2703569378155ebc4ebeea897b0742ef6cc66d6b))
* use user as the filter value instead of modified (#16586) ([94f28d7](https://github.com/ParsimonyGit/frappe/commit/94f28d776dfb432da250ae2837a2aa0d7b7bcd8f))
*  **desk:** Use Int control for Long Int (#16572) ([8905551](https://github.com/ParsimonyGit/frappe/commit/8905551994a8825fb552248b7ca8eeb6f2882428))
* Email Dialog 'To' field is not full width (backport #16603) (#16606) ([905fd99](https://github.com/ParsimonyGit/frappe/commit/905fd99b5fff8ca851c957be532e98b7f7be35c8))
* Don't update saved filters "silently" ([b143155](https://github.com/ParsimonyGit/frappe/commit/b143155843fefa94347b906d258b048794512e42))


### Chore

* better variable name ([b6a72a7](https://github.com/ParsimonyGit/frappe/commit/b6a72a78ad0595fd68f58c473a28261654892287))
* ignore bulk formatting commit from git-blame ([5cade34](https://github.com/ParsimonyGit/frappe/commit/5cade342cc9d2d1c28e1b371620448f9d06088a9))
* change log and version bump ([5caf78d](https://github.com/ParsimonyGit/frappe/commit/5caf78dc603587f1ad26bad684385e20f0f2db4f))
* Add change log ([ba69e77](https://github.com/ParsimonyGit/frappe/commit/ba69e77e397747c328f676f9a27c450b9fed3c98))


### Styles

* Added gap between from and to field ([85d3c41](https://github.com/ParsimonyGit/frappe/commit/85d3c41cd067fd316e5f842d0a63d7d03be5c4d1))
* Format codebase & sort imports ([b43f19f](https://github.com/ParsimonyGit/frappe/commit/b43f19f2dfa532a6c64fc5ed6b9704962e0f9a00))


### Continuous Integration

* Add isort & black to pre-commit config ([d8be72d](https://github.com/ParsimonyGit/frappe/commit/d8be72d261908a9a82a0580740d214243198e6cf))


### Contributors

<a href="https://github.com/mergify">mergify[bot]</a>, <a href="https://github.com/nabinhait">nabinhait</a>, <a href="https://github.com/phot0n">phot0n</a>, <a href="https://github.com/shariquerik">shariquerik</a>, <a href="https://github.com/surajshetty3416">surajshetty3416</a>, <a href="https://github.com/frappe-bot">frappe-bot</a>, <a href="https://github.com/hans-permana">hans-permana</a>, <a href="https://github.com/gavindsouza">gavindsouza</a>, <a href="https://github.com/deepeshgarg007">deepeshgarg007</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

Commit:
test: fix git authentication

mergify[bot] committed:
fix: Grid layout broken for some grids (#16702) (#16716)

mergify[bot] committed:
feat: more color options for form alerts (backport #16729) (#16738)

* feat: more color options for form alerts (#16729)

* Added green and orange color options for message

(cherry picked from commit 2767ea9d255b9f8c8a2e3840cf7b2df50ce36b56)

# Conflicts:
#	frappe/public/js/frappe/form/layout.js

* chore: conflicts

Co-authored-by: Nikhil Kothari <nik.kothari22@live.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
test: flaky UI tests (backport #16718) (#16746)

deepeshgarg007 committed:
Merge pull request #16750 from frappe/version-13-hotfix

chore: version 13 pre release

deepeshgarg007 committed:
Merge pull request #16751 from frappe/version-13-pre-release

chore: version 13 release

deepeshgarg007 committed:
chore: failing semantic release

(cherry picked from commit 59ee952d82548b1616ce79349487bd4210dd5863)

deepeshgarg007 committed:
chore: failing semantic release

(cherry picked from commit 59ee952d82548b1616ce79349487bd4210dd5863)

deepeshgarg007 committed:
chore: failing semantic release

(cherry picked from commit 59ee952d82548b1616ce79349487bd4210dd5863)

deepeshgarg007 committed:
Merge pull request #16753 from frappe/mergify/bp/version-13-hotfix/pr-16752

chore: failing semantic release (backport #16752)

deepeshgarg007 committed:
Merge pull request #16754 from frappe/mergify/bp/version-13-pre-release/pr-16752

chore: failing semantic release (backport #16752)

deepeshgarg007 committed:
Merge pull request #16755 from frappe/mergify/bp/version-13/pr-16752

chore: failing semantic release (backport #16752)

frappe-bot committed:
chore(release): Bumped to Version 13.27.0

# [13.27.0](https://github.com/frappe/frappe/compare/v13.26.1...v13.27.0) (2022-04-26)

### Bug Fixes

* Check mandatory_depends_on also while checking mandatory on save ([#16515](https://github.com/frappe/frappe/issues/16515)) ([#16686](https://github.com/frappe/frappe/issues/16686)) ([e4e2090](https://github.com/frappe/frappe/commit/e4e20904ba805efc3950d49f11221dc5cbaac96c))
* Do not change direction of text if already in RTL mode ([9c3a488](https://github.com/frappe/frappe/commit/9c3a4883e9bb910cc0222127ee4b3f5a924ca5f6))
* Grid layout broken for some grids ([#16702](https://github.com/frappe/frappe/issues/16702)) ([#16716](https://github.com/frappe/frappe/issues/16716)) ([04c1170](https://github.com/frappe/frappe/commit/04c117073184061f22cc82cdae5c30ac263d7497))
* handle no data in set_open_count (backport [#16582](https://github.com/frappe/frappe/issues/16582)) ([#16696](https://github.com/frappe/frappe/issues/16696)) ([ca91813](https://github.com/frappe/frappe/commit/ca918139ea9562ef03b6e4e558fcca43fef97de9))
* Handle unable to read commit errors ([92a2e97](https://github.com/frappe/frappe/commit/92a2e97a0e93811700345aeda3110738153be46d))
* **UI:** Duration Filter overlapping issue  (backport [#16599](https://github.com/frappe/frappe/issues/16599)) ([#16703](https://github.com/frappe/frappe/issues/16703)) ([c997ad9](https://github.com/frappe/frappe/commit/c997ad9f0796cf5902430e3a7fb3058e54654c68))

### Features

* add/remove fields from kanban board (backport [#16257](https://github.com/frappe/frappe/issues/16257)) ([#16671](https://github.com/frappe/frappe/issues/16671)) ([bf63adb](https://github.com/frappe/frappe/commit/bf63adbfee69b3c9001f975f06d38da5190128e8))
* configurable auto email reports limit ([#16684](https://github.com/frappe/frappe/issues/16684)) ([0836cdc](https://github.com/frappe/frappe/commit/0836cdc84be73781163ea18f269050bf967097d2))
* **minor:** add ignore_xss_filter to customize form field (backport [#16612](https://github.com/frappe/frappe/issues/16612)) ([#16674](https://github.com/frappe/frappe/issues/16674)) ([8a0b805](https://github.com/frappe/frappe/commit/8a0b8052fbe1aafb57c0ed7147072cef1703ee9f))
* more color options for form alerts (backport [#16729](https://github.com/frappe/frappe/issues/16729)) ([#16738](https://github.com/frappe/frappe/issues/16738)) ([0c92013](https://github.com/frappe/frappe/commit/0c9201356a376f07b764184052fb38ed6ff19734))

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

hrwX committed:
feat: use mobile native camera (#16709)

shadrak98 committed:
fix: properly validate google sheets url (#16683)

* fix: properly validate google sheets url

* fix: check for spreadsheets in path

* fix(minor): error should throw if any of the cond fails

(cherry picked from commit e0c89cdc7375092e150bfbbbd61288cc7099ae7c)

saurabh6790 committed:
feat: provision to handle payment authorization event in server script for custom documents (#16712)

Currently, there is no provision to handle payment authorization events via server script. So it's not possible if a user wants to link payments against custom documents.

Thus adding a provision in server script
- Setup checkout for custom doc
<img width="1299" alt="Screenshot 2022-03-11 at 2 44 19 PM" src="https://user-images.githubusercontent.com/3784093/157838010-989f6009-83d6-4329-92a6-bb2b6b38438b.png">

- Handle payment callback
<img width="1051" alt="Screenshot 2022-04-22 at 11 28 19 AM" src="https://user-images.githubusercontent.com/3784093/164612613-d321edb3-c7f8-4063-b809-8ae5fe46ee66.png">

## Documentation
https://frappeframework.com/docs/v13/user/en/desk/scripting/server-script/edit?wiki_page_patch=bbed0fcd9a
(cherry picked from commit e2d3d1d0be53fd09409febf038159805c27ec5b3)

barredterra committed:
fix: Replace password with asterisks before logging (#16743)

(cherry picked from commit 5d61482fb35cba4dba8032aa6c14092e28bcff88)

mergify[bot] committed:
fix: Newsletter unsubscribe button not working (backport #16756) (#16760)

gavindsouza committed:
fix(mobile): Show checkbox in file list view

(cherry picked from commit 0551a2f1e7b80c2f5e3ae16ba579aa7d2d063674)

gavindsouza committed:
Merge pull request #16761 from frappe/mergify/bp/version-13-hotfix/pr-16743

fix: Replace password with asterisks before logging (backport #16743)

gavindsouza committed:
Merge pull request #16764 from frappe/mergify/bp/version-13-hotfix/pr-16762

fix(mobile): Show checkbox in file list view (backport #16762)

barredterra committed:
refactor: Sync fixtures

(cherry picked from commit 8f53a039a6665d3e4cda28f1075ab41cb7761bda)

# Conflicts:
#	frappe/utils/fixtures.py

surajshetty3416 committed:
fix(newsletter): Pass parent instead of parentfield

while getting successful recipients of a newsletter

(cherry picked from commit 2c4773ff4482ed3a2902846c886a7a16905f11a8)

saurabh6790 committed:
chore: add missing doc event type

surajshetty3416 committed:
Merge pull request #16779 from frappe/mergify/bp/version-13-hotfix/pr-16774

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-16712

surajshetty3416 committed:
Merge pull request #16759 from frappe/mergify/bp/version-13-hotfix/pr-16712

barredterra committed:
fix: resolve merge conflicts

gavindsouza committed:
Merge pull request #16776 from frappe/mergify/bp/version-13-hotfix/pr-16773

refactor: Sync fixtures (backport #16773)

mergify[bot] committed:
fix: console error while using TAB shortcut in grid (#16701) (#16785)

mergify[bot] committed:
ci: failfast in case of conflicts (#16777) (#16781)

(cherry picked from commit 9bec3480ed90376f12052b2933123aeab5a456ce)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: dont throw error for a bad translation (backport #16769) (#16775)

* fix: dont throw error for a bad translation (#16769)

(cherry picked from commit b714dbb190311b43f22c898d5925d045b4150a62)

# Conflicts:
#	frappe/translate.py

* chore: conflitcs

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
test: clear DB value cache (#16794)

hrwX committed:
feat: disable change log

(cherry picked from commit 63e2e328c6c0a1d350b23be4fd13408d54b936a6)

mergify[bot] committed:
fix: use backticks for fieldname while preparing filters (backport #16471) (#16708)

* fix: use backticks for fieldname while preparing filters

* test: test for fieldname which start with int

* chore: resolve conflicts

* fix: use data fieldtype instead of int

Co-authored-by: phot0n <ritwikpuri5678@gmail.com>

gavindsouza committed:
fix: Use BlogPosting schema for Article

Fixes https://github.com/frappe/frappe/issues/1730

(cherry picked from commit 6d96f11dd961d7f71cd54cae421793209c86f921)

surajshetty3416 committed:
Merge pull request #16801 from frappe/mergify/bp/version-13-hotfix/pr-16800

mergify[bot] committed:
fix: use proper validations for number card validation method (#16581) (#16803)

(cherry picked from commit 76a6c282b81dfdc81ebf64a1e5cb9bdc0a5b15d6)

Co-authored-by: Ritwik Puri <ritwikpuri5678@gmail.com>

mergify[bot] committed:
fix(UX): suggest app-specific password for gmail logins (#16812) (#16813)

(cherry picked from commit 4310a2ecca213193a42065ccde0c7d2fbbd4a807)

Co-authored-by: Ankush Menat <ankush@frappe.io>

deepeshgarg007 committed:
Merge pull request #16820 from frappe/version-13-hotfix

chore: pre release for version-13

deepeshgarg007 committed:
Merge pull request #16821 from frappe/version-13-pre-release

chore: weekly release for version-13

frappe-bot committed:
chore(release): Bumped to Version 13.28.0

# [13.28.0](https://github.com/frappe/frappe/compare/v13.27.0...v13.28.0) (2022-05-03)

### Bug Fixes

* console error while using TAB shortcut in grid ([#16701](https://github.com/frappe/frappe/issues/16701)) ([#16785](https://github.com/frappe/frappe/issues/16785)) ([b2960b9](https://github.com/frappe/frappe/commit/b2960b91f435039902ca670a30ce12cae345294a))
* dont throw error for a bad translation (backport [#16769](https://github.com/frappe/frappe/issues/16769)) ([#16775](https://github.com/frappe/frappe/issues/16775)) ([d34292e](https://github.com/frappe/frappe/commit/d34292e9cb4fc58a19ac9bddb891318703f36cfa))
* **mobile:** Show checkbox in file list view ([5fa3d29](https://github.com/frappe/frappe/commit/5fa3d295dc01673b30d11fe89e778760a71327db))
* Newsletter unsubscribe button not working (backport [#16756](https://github.com/frappe/frappe/issues/16756)) ([#16760](https://github.com/frappe/frappe/issues/16760)) ([b4392e1](https://github.com/frappe/frappe/commit/b4392e1b9fcb1a8973c0879a9fd7cc3a8663dd89))
* **newsletter:** Pass parent instead of parentfield ([defc26e](https://github.com/frappe/frappe/commit/defc26e0a6fec1e656b2427fffa1b4583cf15854))
* Replace password with asterisks before logging ([#16743](https://github.com/frappe/frappe/issues/16743)) ([559d7cf](https://github.com/frappe/frappe/commit/559d7cf6826f3a1ae1b669e7659ac061eb4abc1e))
* resolve merge conflicts ([93c70ee](https://github.com/frappe/frappe/commit/93c70eea082641f12cbc784a045ddc4b5a0ef5c1))
* use backticks for fieldname while preparing filters (backport [#16471](https://github.com/frappe/frappe/issues/16471)) ([#16708](https://github.com/frappe/frappe/issues/16708)) ([a38a2d0](https://github.com/frappe/frappe/commit/a38a2d02f2ef52c252960aa18f94420a230b2abf))
* Use BlogPosting schema for Article ([f4c7ebd](https://github.com/frappe/frappe/commit/f4c7ebda41ec10fcea1eb604b412f0ce61ea2a3c))
* use proper validations for number card validation method ([#16581](https://github.com/frappe/frappe/issues/16581)) ([#16803](https://github.com/frappe/frappe/issues/16803)) ([d7c4710](https://github.com/frappe/frappe/commit/d7c4710ac40aabc409c02e2a23f4d0e073ee6bba))
* **UX:** suggest app-specific password for gmail logins ([#16812](https://github.com/frappe/frappe/issues/16812)) ([#16813](https://github.com/frappe/frappe/issues/16813)) ([81d56f9](https://github.com/frappe/frappe/commit/81d56f97ed1aa73ff4f11b7f1048b2522c4dcbf2))

### Features

* provision to handle payment authorization event in server script for custom documents ([#16712](https://github.com/frappe/frappe/issues/16712)) ([c81fb61](https://github.com/frappe/frappe/commit/c81fb61bae4151c5b564d37cf110b22c72fc9ce2))
* use mobile native camera ([#16709](https://github.com/frappe/frappe/issues/16709)) ([86bbc43](https://github.com/frappe/frappe/commit/86bbc43045aa947fee45fada95d9c76230b07c5f))

mergify[bot] committed:
perf: remove naming series from log-like doctypes (#16823) (#16826)

- webhook request log
- access log

(cherry picked from commit d1938ee27198493c58df065a6111635a6340fad6)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: bad query if user has ' in the email address (backport #16796) (#16834)

ankush committed:
Merge pull request #16845 from frappe/mergify/bp/version-13-hotfix/pr-16683

fix: properly validate google sheets url (backport #16683)

mergify[bot] committed:
fix: frappe.throw in case of list of errors (#16855) (#16857)

(cherry picked from commit 61223092e0386016397ec738221aca7855d3e746)

Co-authored-by: Saqib Ansari <nextchamp.saqib@gmail.com>

surajshetty3416 committed:
Merge pull request #16861 from frappe/mergify/bp/version-13-hotfix/pr-16797

gavindsouza committed:
fix(dashboard): Consistent "No Data" message in all charts

(cherry picked from commit 7098f29dfea6d4812dad4be2341b01bfd5d87473)

gavindsouza committed:
fix(dashboard): Don't show stats in number widget if NaN

(cherry picked from commit b618d29296732d096c5d74885a25bc4251981024)

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

deepeshgarg007 committed:
Merge pull request #16864 from frappe/version-13-hotfix

chore: Pre release for version-13

deepeshgarg007 committed:
Merge pull request #16865 from frappe/version-13-pre-release

chore: Release for version-13

frappe-bot committed:
chore(release): Bumped to Version 13.29.0

# [13.29.0](https://github.com/frappe/frappe/compare/v13.28.0...v13.29.0) (2022-05-10)

### Bug Fixes

* bad query if user has ' in the email address (backport [#16796](https://github.com/frappe/frappe/issues/16796)) ([#16834](https://github.com/frappe/frappe/issues/16834)) ([6cb2955](https://github.com/frappe/frappe/commit/6cb2955386878652f15cee5bed9959b321cae465))
* frappe.throw in case of list of errors ([#16855](https://github.com/frappe/frappe/issues/16855)) ([#16857](https://github.com/frappe/frappe/issues/16857)) ([27a9d2d](https://github.com/frappe/frappe/commit/27a9d2d1fa5f75605db2219854b4a590796f002b))
* properly validate google sheets url ([#16683](https://github.com/frappe/frappe/issues/16683)) ([f0d3a02](https://github.com/frappe/frappe/commit/f0d3a027ef838e5d02b2d5c82b88eb8ffe41a841))

### Features

* disable change log ([1897af2](https://github.com/frappe/frappe/commit/1897af25354e205ee10f6e8520fc947c09dc965a))

### Performance Improvements

* remove naming series from log-like doctypes ([#16823](https://github.com/frappe/frappe/issues/16823)) ([#16826](https://github.com/frappe/frappe/issues/16826)) ([bf2d6cd](https://github.com/frappe/frappe/commit/bf2d6cdbda3e6b5127f6f4b050aa8fc42b29e1cb))

gavindsouza committed:
chore: Simplify get_open_count doc usage

(cherry picked from commit c16bec3407328ae14e07d1e1c5f8f3419849bda6)

gavindsouza committed:
fix: Fetch meta to ensure formatting for Dashboard Charts

* This handles setting meta for dashboard chart of types: Sum, Count,
  Average, Group By
* Dhashboard Charts of type Report have to pass fieldtype in chart data
  Dict to apply formatting. This can't be done magically AFAIK

(cherry picked from commit 66ccd103ed1504c31da29a8bdb691d7268702835)

hrwX committed:
fix: change ordering to creation (#16863)

(cherry picked from commit 5ac729252f3319dec35608be5c39c305afdf02f5)

# Conflicts:
#	frappe/contacts/address_and_contact.py

gavindsouza committed:
fix: No optional chaining for you v13

Based on SyntaxErrors on rollup build :')

gavindsouza committed:
Merge pull request #16874 from frappe/mergify/bp/version-13-hotfix/pr-16868

fix: Dashboard view (backport #16868)

surajshetty3416 committed:
fix: Resolve conflicts

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

surajshetty3416 committed:
Merge pull request #16870 from frappe/mergify/bp/version-13-hotfix/pr-16863

CleoMenezesJr committed:
fix: #16873 - broken indicator-pills after color hex code fix

As stated in issue, commit 4ff474d0883b356bc498cb62e7e4ff591e387429 changed the way indicator-pills are colored, and it looks broken.

surajshetty3416 committed:
Merge pull request #16881 from CleoMenezesJr/version-13-hotfix

revert: "fix: color hex code not work as class name"

mergify[bot] committed:
fix: recursion error in translations (backport #16894) (#16896)

* fix: recursion error in translations

If bad translations is found then while `msgprint` it attempts to load
translation again because of its arg `title = _("Message")`

(cherry picked from commit 55def8b7186f6879481462ad22e034b1063b654c)

* test: verify ALL translation file syntax

(cherry picked from commit 26b54221ea546bfbf52d4927beb33a003c327aee)

Co-authored-by: Ankush Menat <me@ankush.dev>

finbyz committed:
fix: Filters Won't Work While Routing in Report (#16895)

mergify[bot] committed:
fix: add dependent status field in reportview (backport #16903) (#16906)

ankush committed:
Merge pull request #16916 from ankush/v13_release

chore: release

frappe-bot committed:
chore(release): Bumped to Version 13.29.1

## [13.29.1](https://github.com/frappe/frappe/compare/v13.29.0...v13.29.1) (2022-05-17)

### Bug Fixes

* [#16873](https://github.com/frappe/frappe/issues/16873) - broken indicator-pills after color hex code fix ([2daa422](https://github.com/frappe/frappe/commit/2daa422ec754541776395ad8715186b3b738b532))
* add dependent status field in reportview (backport [#16903](https://github.com/frappe/frappe/issues/16903)) ([#16906](https://github.com/frappe/frappe/issues/16906)) ([dca7781](https://github.com/frappe/frappe/commit/dca778147360e9bd0f20e8ddc80b4fedc0a20a85))
* change ordering to creation ([#16863](https://github.com/frappe/frappe/issues/16863)) ([afb5397](https://github.com/frappe/frappe/commit/afb5397189b5be120eeeb5bbd5a6327144bdeb34))
* **dashboard:** Consistent "No Data" message in all charts ([26ad877](https://github.com/frappe/frappe/commit/26ad8772573fd1142b33089b871505fbb07a457b))
* **dashboard:** Don't show stats in number widget if NaN ([97932c0](https://github.com/frappe/frappe/commit/97932c025b290aa1dd7a2cc84a642ded6e0cfa93))
* Fetch meta to ensure formatting for Dashboard Charts ([c1ba739](https://github.com/frappe/frappe/commit/c1ba7396cf79aa7600e6c8959ae8dc1f8b4a4f4b))
* Filters Won't Work While Routing in Report ([#16895](https://github.com/frappe/frappe/issues/16895)) ([eef9db1](https://github.com/frappe/frappe/commit/eef9db10cd0cb5bee85178542dae51b28b6b9e00))
* No optional chaining for you v13 ([6024000](https://github.com/frappe/frappe/commit/60240008a89568b89135a14e14bf5f42dc568475))
* recursion error in translations (backport [#16894](https://github.com/frappe/frappe/issues/16894)) ([#16896](https://github.com/frappe/frappe/issues/16896)) ([bf14a8c](https://github.com/frappe/frappe/commit/bf14a8cd971b6a62edddbb80d78081e86a39a777))
* Resolve conflicts ([1412802](https://github.com/frappe/frappe/commit/1412802ef919cae8da530a31992a0cab35e87b81))

mergify[bot] committed:
feat: Add font sizes dropdown in text editor(quill) (backport #16910) (#16915)

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

surajshetty3416 committed:
fix: Check if email account exists before picking signature

surajshetty3416 committed:
Merge pull request #16922 from surajshetty3416/fix-email-signature-filter

mergify[bot] committed:
fix: Passing `dev server` variable to jenv globals correctly (backport #16843) (#16923)

* fix: Passing `dev server` variable to jenv globals correctly (#16843)

Encountered the same issue during a fresh setup: https://discuss.erpnext.com/t/v13-fresh-install-develop-socket-io-using-wrong-port/74562

The problem originates from the `socketio_client.js` file.  During development, the `socket.io` client by default sends request to port 8000 but ideally it should've been 9000 (that's where it's actually running) since the latter is defined in `frappe.boot.socketio_port`. This is happening because the `jenv` variable `dev_server` is always being returned as 0.

In [this](https://github.com/frappe/frappe/blob/0f7d8e790339868291d9774b7a9e48b230be781f/frappe/__init__.py#L237) line, we are seting `frappe.local.dev_server` but in safe exec we are using `frappe._dev_server` as the source to populate the variable `dev_server` which eventually gets pushed to the `app.html` via `jinja`. (`frappe._dev_server` evaluates to `None`)

In `socketio_client`, we are using [this](https://github.com/frappe/frappe/blob/0f7d8e790339868291d9774b7a9e48b230be781f/frappe/public/js/frappe/socketio_client.js#L112) variable as a flag to decide whether or not `frappe.boot.socketio_port` should be the port that the `socket.io` client connects to.

(cherry picked from commit 378149375d9ed1e873894dce308e8fdadd8b2141)

# Conflicts:
#	frappe/__init__.py

* fix: Accept current changes - resolve merge conflicts

Co-authored-by: Abhishek M <nateriver210@gmail.com>
Co-authored-by: gavin <gavin18d@gmail.com>

mergify[bot] committed:
fix: allow guests/users without desk access to upload text files (backport #16920) (#16926)

fix: allow guests/users without desk access to upload text files (#16920)

* fix: allow guests/users without desk access to upload text files

* chore: update error message

Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>
(cherry picked from commit 20d73cae24f40a91a0d1d3409ff63273ae3de377)

Co-authored-by: Ritwik Puri <ritwikpuri5678@gmail.com>

mergify[bot] committed:
fix: remove old test runner doc (backport #16929) (#16930)

fix: remove old test runner doc (#16929)

Not sure from where this ever came but on some old site it's present and
causes installation issue.

closes https://github.com/frappe/frappe/issues/16917

(cherry picked from commit e8657d52ed5b3f68af32e1b61381174c72c5cecd)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
fix: Dashboard charts not loading (#16921)

surajshetty3416 committed:
Merge pull request #16933 from frappe/mergify/bp/version-13-hotfix/pr-16921

fix: Dashboard charts not loading (backport #16921)

ron-ulitsky committed:
Update importer.py

fixed bug that returned incorrect extensions for file paths with directories containing full stops

(cherry picked from commit 65b1474cadb7f9b7401405456c62512d366ae329)

surajshetty3416 committed:
Merge pull request #16937 from frappe/mergify/bp/version-13-hotfix/pr-16936

surajshetty3416 committed:
fix(form): Use actual doctype of link intead of parent doctype

Due to this wrong doctype, Link Preview popover was not working as expected.

(cherry picked from commit bb91907fd05bc8b4fc105f9f27c4891f69ebbbd6)

surajshetty3416 committed:
Merge pull request #16940 from frappe/mergify/bp/version-13-hotfix/pr-16939

fix(form): Set actual doctype of link instead of parent doctype (backport #16939)

mergify[bot] committed:
fix: Dashboard charts not loading (backport #16921) (#16943)

fix: Dashboard charts not loading (#16921)

(cherry picked from commit 0a76a9314b6ea2d07854ca5012ec437dcc73c3ac)

Co-authored-by: Ankush Menat <ankush@frappe.io>

frappe-bot committed:
chore(release): Bumped to Version 13.29.2

## [13.29.2](https://github.com/frappe/frappe/compare/v13.29.1...v13.29.2) (2022-05-20)

### Bug Fixes

* Dashboard charts not loading (backport [#16921](https://github.com/frappe/frappe/issues/16921)) ([#16943](https://github.com/frappe/frappe/issues/16943)) ([ef9b833](https://github.com/frappe/frappe/commit/ef9b833864274cc976d7c8d181f9f5ce6aab3c03))

mergify[bot] committed:
fix: use reference to form instead of global var (backport #16942) (#16944)

fix: use reference to form instead of global var (#16942)

(cherry picked from commit 6e53c3d339aeed42232dbd7f9fac4b5d5973a784)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
perf: ~3x faster server side `_(translate)` calls (backport #16949) (#16951)

perf: dont rebuild same translation dict

(cherry picked from commit d90d6536ef281a0f4c46d6ac4c724672f16a9a34)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
perf(translation): improve if condition to return all values except `None` (backport #16953)

perf(translation): improve if condition to return all values except `None` (#16953)

(cherry picked from commit 97df77f9541894a8dd55f62522f304d8e5e8c060)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

barredterra committed:
feat: Add german translations

(cherry picked from commit 88f65cdefbfead538f64e8788ee7c70495151d96)

barredterra committed:
feat: Add german translations

(cherry picked from commit d83251ad474d14863f3e1be0886f03c498d26e17)

FHenry committed:
feat: if only on email per user the preselect it in email form

(cherry picked from commit e41bad6066290fecf7b9ba3d76ce1132a146ca98)

surajshetty3416 committed:
Merge pull request #16965 from frappe/mergify/bp/version-13-hotfix/pr-16964

feat: if only one email per user then preselect it in email form (backport #16964)

surajshetty3416 committed:
Merge pull request #16966 from frappe/mergify/bp/version-13-hotfix/pr-16962

feat: Add german translations (backport #16962)

mergify[bot] committed:
ci: `Skip CI` label for not running build processes (backport #16902) (#16967)

* ci: Skip CI label for stopping build process

(cherry picked from commit 909ce30f0a10e244e3ce1bc74aa4498c301f10c7)

* chore: fix conflict

Co-authored-by: phot0n <ritwikpuri5678@gmail.com>

mergify[bot] committed:
fix: delete email queue first (backport #16968) (#16970)

fix: delete email queue first (#16968)

email queue takes a long time and during that previously deleted tables
remain locked. Activity/Error logs are relatively faster to delete so
should moved last

(cherry picked from commit bfda3869a6726f3fe436508b117da4c20e880264)

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
fix: hide slideshow if page builder is selected (#16975)

mergify[bot] committed:
fix: mention only users that can be read (backport #16978) (#16980)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
fix: update french translation (backport #16974) (#16986)

fix: update franch translation (#16974)

remove plurals (as other translation)

(cherry picked from commit 3a05c0f0933c0b8012b58cf2a8b22566ab81b4e6)

Co-authored-by: HENRY Florian <florian.henry@scopen.fr>

mergify[bot] committed:
feat: Display button "Create User Email" only if current user can create an Email Account (backport #16979) (#16992)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
Fix: Change currency fraction for Singapore from Sen to Cent (backport #16987) (#16991)

Co-authored-by: Byju Abraham <65185610+polycurve@users.noreply.github.com>

deepeshgarg007 committed:
Merge pull request #16996 from frappe/version-13-hotfix

chore: Weekly release for version-13

frappe-bot committed:
chore(release): Bumped to Version 13.30.0

# [13.30.0](https://github.com/frappe/frappe/compare/v13.29.2...v13.30.0) (2022-05-26)

### Bug Fixes

* allow guests/users without desk access to upload text files (backport [#16920](https://github.com/frappe/frappe/issues/16920)) ([#16926](https://github.com/frappe/frappe/issues/16926)) ([82f7593](https://github.com/frappe/frappe/commit/82f7593b7857894cce127d57934da2465ede4a57))
* Check if email account exists before picking signature ([2e5bd71](https://github.com/frappe/frappe/commit/2e5bd71d95c1ab2d5d089945da6065b1a059ae59))
* Dashboard charts not loading ([#16921](https://github.com/frappe/frappe/issues/16921)) ([0a76a93](https://github.com/frappe/frappe/commit/0a76a9314b6ea2d07854ca5012ec437dcc73c3ac))
* delete email queue first (backport [#16968](https://github.com/frappe/frappe/issues/16968)) ([#16970](https://github.com/frappe/frappe/issues/16970)) ([a9135f8](https://github.com/frappe/frappe/commit/a9135f8e0bca7c600e70ebf6558a917cc6ea93bb))
* **form:** Use actual doctype of link intead of parent doctype ([a771292](https://github.com/frappe/frappe/commit/a771292c6977afed36f1e7cf20b03328862c2c23))
* hide slideshow if page builder is selected ([#16975](https://github.com/frappe/frappe/issues/16975)) ([5c3b589](https://github.com/frappe/frappe/commit/5c3b589e5b1ea012c0eaef7599d734be9c1b06d3))
* mention only users that can be read (backport [#16978](https://github.com/frappe/frappe/issues/16978)) ([#16980](https://github.com/frappe/frappe/issues/16980)) ([a0b5deb](https://github.com/frappe/frappe/commit/a0b5deb34a6c048372086c996682574a2197a445))
* Passing `dev server` variable to jenv globals correctly (backport [#16843](https://github.com/frappe/frappe/issues/16843)) ([#16923](https://github.com/frappe/frappe/issues/16923)) ([23e1fc3](https://github.com/frappe/frappe/commit/23e1fc39350bf9e43a87d9ffd4df721734b175e2)), closes [/github.com/frappe/frappe/blob/0f7d8e790339868291d9774b7a9e48b230be781f/frappe/__init__.py#L237](https://github.com//github.com/frappe/frappe/blob/0f7d8e790339868291d9774b7a9e48b230be781f/frappe/__init__.py/issues/L237) [/github.com/frappe/frappe/blob/0f7d8e790339868291d9774b7a9e48b230be781f/frappe/public/js/frappe/socketio_client.js#L112](https://github.com//github.com/frappe/frappe/blob/0f7d8e790339868291d9774b7a9e48b230be781f/frappe/public/js/frappe/socketio_client.js/issues/L112)
* remove old test runner doc (backport [#16929](https://github.com/frappe/frappe/issues/16929)) ([#16930](https://github.com/frappe/frappe/issues/16930)) ([9067123](https://github.com/frappe/frappe/commit/906712380342d8e8e1b43fdd4ee3b4f93f87d338))
* update french translation (backport [#16974](https://github.com/frappe/frappe/issues/16974)) ([#16986](https://github.com/frappe/frappe/issues/16986)) ([a3915c2](https://github.com/frappe/frappe/commit/a3915c289a4569884bd5aea0ed567324a9880b70))
* use reference to form instead of global var (backport [#16942](https://github.com/frappe/frappe/issues/16942)) ([#16944](https://github.com/frappe/frappe/issues/16944)) ([29d14e0](https://github.com/frappe/frappe/commit/29d14e073db2be2877002a335bdc563583508703))

### Features

* Add font sizes dropdown in text editor(quill) (backport [#16910](https://github.com/frappe/frappe/issues/16910)) ([#16915](https://github.com/frappe/frappe/issues/16915)) ([8a3ec86](https://github.com/frappe/frappe/commit/8a3ec866a8522cecba5532bed1b11bc14b6253a0))
* Add german translations ([88c1f83](https://github.com/frappe/frappe/commit/88c1f832d3a37e1ec9b5e06040f9752fd363ace1))
* Add german translations ([36f6cb9](https://github.com/frappe/frappe/commit/36f6cb9a383511d1535e9da657f8dbfb14f899b8))
* Display button "Create User Email" only if current user can create an Email Account (backport [#16979](https://github.com/frappe/frappe/issues/16979)) ([#16992](https://github.com/frappe/frappe/issues/16992)) ([2ec5622](https://github.com/frappe/frappe/commit/2ec5622c59aedae9d2fdea8138993deee6de183b))
* if only on email per user the preselect it in email form ([47de126](https://github.com/frappe/frappe/commit/47de126f320a3ee97a3507756d2e6bd9ae6fd025))

### Performance Improvements

* ~3x faster server side `_(translate)` calls (backport [#16949](https://github.com/frappe/frappe/issues/16949)) ([#16951](https://github.com/frappe/frappe/issues/16951)) ([1de2432](https://github.com/frappe/frappe/commit/1de2432453c82f3837ecf082b1495d80616c6698))
* **translation:** improve if condition to return all values except `None` (backport [#16953](https://github.com/frappe/frappe/issues/16953)) ([bd04589](https://github.com/frappe/frappe/commit/bd045893a7dfbc9efe7f8b297c0ca78480bb9f4e))

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13
